custom/icds/const.py

```diff
@@ -32,3 +32,5 @@
 FILE_TYPE_CHOICE_DOC = 2
 DISPLAY_CHOICE_LIST = 1
 DISPLAY_CHOICE_FOOTER = 2
+
+DATA_PULL_CACHE_KEY = "icds_custom_data_pull_in_progress"
```

Comment:
I think this whole commit might be able to be replaced with a celery `@serial_task` decorator (though that wouldn't prevent people from running the management command multiple times...)

Reply:
we would want any data pull run from anywhere to be prevented if one is already running.

Reply:
Ya, need to prevent any parallel data pull